src/movies.js

```diff
@@ -1,25 +1,99 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    let directorsArr = []
+    const directorMap = moviesArray.map((movie) => {
+        directorsArr.push(movie.director)
```

Comment:
Map gives you a new array, don't need to make a new one and push elements inside.
```javascript
function getAllDirectors(moviesArray) {
    let allDirectors = moviesArray.map(movie =>{
        return movie.director;
    });
    return allDirectors;
}
```